pkg/sql/logictest/testdata/logic_test/alter_type

```diff
@@ -666,3 +666,32 @@ CREATE VIEW bar_view AS (SELECT ARRAY['b'::bar])
 
 statement error pq: could not remove enum value "b" as it is being used in view "bar_view"
 ALTER TYPE bar DROP VALUE 'b'
+
+subtest regression_97975
+
+statement ok
+CREATE DATABASE db97975;
+USE db97975;
+
+statement ok
+CREATE TYPE enum_97975 as ENUM ('value1', 'value2');
+
+query T
```

Comment:
Can you see what happens when there are multiple user sessions with temp tables using the same enum?

Reply:
This is a really good call. And unfortunately this pr probably doesn't quite work.
1. Looks like sessions can see temp tables from other session. At least in `planner.forEachMutableTableInDatabase` method, you actually can get mutable temp tables from another session.
2. When this method is called from jobs, looks like we don't pass down session ids :(

Reply:
Getting the temp schema name from namespace table should be fine though.
But I'm not sure how the cross session behavior should be. In theory, I think a session adding an enum should make corresponding changes to temp table (if it uses the enum) in another session. But this can be surprising to users.

Reply:
(Bug reporter here) I think that the lesser evil would be to have the new value be visible to all session temp tables, in that to do the otherwise would be more surprising. What other system-wide schema changes would be shielded in this fashion?

The likelihood that other sessions will notice or care about the new enum value is real low. That said, PostgreSQL behaves this way in read-committed transaction isolation:

```
defaultdb session 1=# create type my_type as enum ('value_1', 'value_2');
CREATE TYPE
defaultdb session 1=# create temporary table temp_session_1 (value my_type not null);
CREATE TABLE

defaultdb session 2=# create temporary table temp_session_2 (value my_type not null);
CREATE TABLE

defaultdb session 1=# alter type my_type add value 'value_3';

-- New type value useable in session 2:
defaultdb session 2=# insert into temp_session_2 values ('value_3');
INSERT 0 1
```

The more treacherous corollary comes with  `ALTER TYPE ... DROP VALUE`, but again, the least wrong thing to do regarding temp tables would be to visit every temp table across all sessions.

Reply:
Sorry, just came back to this. Yeah, I agree. Make sense to make new type values visible to other sessions. Thanks for the input.